# Hunting Down Low Identity Cargo Proteins

We have a list of manually curated cargo protein types in `encapsulin_families.csv`. We also have search hits for these cargo proteins against BLAST nr in `cargo_blast_nr_hits.tsv`. Let's comb through it and find some annotated cargo proteins that have low identity to any known protein - or maybe proteins that have homology against proteins that are unannotated or misannotated as something else.

First, we load the BLAST hits:

In [2]:
import pandas as pd

hits_df = pd.read_csv("../cargo_blast_nr_hits.tsv", sep="\t", names=["Query", "Target", "Identity", "Alignment Length", "Mismatches", "Gap Openings",
                                                              "Query Start", "Query End", "Target Start", "Target End", "E-Value", "Bitscore"])

hits_df.head()

Query                                             Target  \
0  MGYP001385583216  MAE56341.1 hypothetical protein [Porticoccacea...   
1  MGYP001385583216  CAB4150874.1 hypothetical protein UFOVP574_44 ...   
2  MGYP001385583216  CAB4171523.1 hypothetical protein UFOVP927_6 [...   
3  MGYP001385583216  MCA8835931.1 hypothetical protein [Pseudomonad...   
4  MGYP001385583216  BAQ84647.1 hypothetical protein [uncultured Me...   

   Identity  Alignment Length  Mismatches  Gap Openings  Query Start  \
0     0.869                84          11             0            4   
1     0.753                77          19             0           11   
2     0.708                79          23             0            9   
3     0.776                67          15             0            8   
4     0.765                64          15             0           24   

   Query End  Target Start  Target End       E-Value  Bitscore  
0         87             2          85  1.157000e-39       152  
1         87             4          80  2.051000e-31       128  
2         87             2          80  1.155000e-28       120  
3         74             2          68  5.500000e-24       107  
4         87            24          87  6.940000e-23       104

Next, let's load a list of annotated encapsulin systems that we've manually curated, and get the cargo proteins for each one:

In [3]:
#Load operon DataFrame and ensure Pfam lists for each cargo MGYP are read correctly
operon_df = pd.read_csv("../operon_df_filtered.csv")
indices = [-10, -9, -8, -7, -6, -5, -4, -3, -2, -1, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
for i in indices:
    operon_df[f"Pfam {i}"] = operon_df[f"Pfam {i}"].apply(eval)

#Load annotated encapsulins
family_df = pd.read_csv("../encapsulin_families.csv")
annotated_encapsulins = family_df["Encapsulin MGYP"].unique()

#Filter out any annotated encapsulins from our operon DataFrame
operon_df = operon_df.query("`Encapsulin MGYP` in @annotated_encapsulins")
operon_df.head()

-10                -9                -8                -7  \
14  MGYP000087665821  MGYP000703964854  MGYP000736392069  MGYP000338316496   
15  MGYP000423818695  MGYP000532926311  MGYP000143698383  MGYP000709860896   
16  MGYP000423818695  MGYP000532926311  MGYP000143698383  MGYP000709860896   
21  MGYP003081621609  MGYP003081621620  MGYP000657744308  MGYP000144496472   
22               NaN               NaN               NaN               NaN   

                  -6                -5                -4                -3  \
14  MGYP002077299657  MGYP000067028803  MGYP000512287377  MGYP000403178503   
15  MGYP000606656294  MGYP000406125095  MGYP000376646145  MGYP000155487403   
16  MGYP000606656294  MGYP000406125095  MGYP000376646145  MGYP000155487403   
21  MGYP003081621650  MGYP000038336433  MGYP000678389341  MGYP002751754157   
22  MGYP003085065548  MGYP000038336433  MGYP000424619035  MGYP001644060519   

                  -2                -1  ...              Pfam 1  \
14  MGYP000081770201  MGYP000173178704  ...  [PF07687, PF01546]   
15  MGYP000482799648  MGYP000241003398  ...           [PF04261]   
16  MGYP000482799648  MGYP000241003398  ...           [PF04261]   
21  MGYP000909261574  MGYP003081648573  ...  [PF14801, PF08704]   
22  MGYP000941113010  MGYP000460280988  ...  [PF08704, PF14801]   

                         Pfam 2                       Pfam 3     Pfam 4  \
14                    [PF10994]                    [PF13419]  [PF00082]   
15            [PF00775, CL0154]  [PF00384, PF04879, PF01568]  [PF10023]   
16            [PF00775, CL0154]  [PF00384, PF04879, PF01568]  [PF10023]   
21  [PF17758, PF16450, PF00004]                           []         []   
22  [PF00004, PF17758, PF16450]                    [PF03136]         []   

       Pfam 5     Pfam 6                                Pfam 7  \
14  [PF01740]  [PF05494]                             [PF04333]   
15  [PF09650]         []  [PF02518, PF00989, PF16927, PF00512]   
16  [PF09650]         []  [PF02518, PF00989, PF16927, PF00512]   
21         []         []                                    []   
22  [PF03136]   [CL0487]                             [PF00977]   

                Pfam 8     Pfam 9            Pfam 10  
14           [PF02470]  [PF02405]          [PF00005]  
15  [PF00072, PF00196]  [PF00072]  [CL0004, PF00884]  
16  [PF00072, PF00196]  [PF00072]  [CL0004, PF00884]  
21                  []         []                 []  
22           [PF13828]  [PF13828]          [PF10825]  

[5 rows x 41 columns]

And finally, let's load our Pfam mapping dictionaries to get descriptions easily:

In [4]:
# The below code will load the Pfam labels dictionary downloaded from the GoogleResearch/Proteinfer GitHub repo
# I've also manually added some code below to fill in some important missing labels that I manually curated
import gzip
import json

with open("../DBs/label_descriptions.json.gz", 'rb') as f:
    with gzip.GzipFile(fileobj=f, mode='rb') as gzip_file:
      labels_dict = json.load(gzip_file)

labels_dict["PF19821"] = "Phage capsid protein"
labels_dict["PF19307"] = "Phage capsid-like protein"
labels_dict['PF19289'] = "PmbA/TldA metallopeptidase C-terminal domain"
labels_dict['PF19290'] = "PmbA/TldA metallopeptidase central domain"
labels_dict['PF20211'] = "Family of unknown function (DUF6571)"
labels_dict['PF19782'] = "Family of unknown function (DUF6267)"
labels_dict['PF19343'] = "Family of unknown function (DUF5923)"
labels_dict['PF20036'] = "Major capsid protein 13-like"
labels_dict['PF18960'] = "Family of unknown function (DUF5702)"
labels_dict['PF18906'] = "Phage tail tube protein"
labels_dict['PF19753'] = "Family of unknown function (DUF6240)"

def get_label(pfam):
    try:
        return(labels_dict[pfam])
    except KeyError:
        return(None)

pfam_df = pd.read_csv("../pfams/cargo_pfams.tsv", sep="\t", names=["Cargo MGYP", "Pfam", "Start", "End"])
pfam_df["Description"] = pfam_df["Pfam"].apply(get_label)

display(pfam_df.head())

Cargo MGYP     Pfam  Start  End  \
0  MGYP004138304253  PF09956     36   83   
1  MGYP003963085681   CL0110    469  743   
2  MGYP004383807105  PF01391      3   56   
3  MGYP003964778019  PF00295     48  153   
4  MGYP003960646311  PF01068    155  384   

                                   Description  
0  Uncharacterized conserved protein (DUF2190)  
1                                         GT-A  
2     Collagen triple helix repeat (20 copies)  
3                Glycosyl hydrolases family 28  
4              ATP dependent DNA ligase domain

Here's some code that will get the encapsulin MGYP for any given cargo MGYP:

In [5]:
cargo_dict = {}

for index, row in operon_df.iterrows():
    for item in row.values[:20]:
        if isinstance(item, str):
            cargo_dict[item] = row[20]

def get_encapsulin(cargo_mgyp):
    try:
        return(cargo_dict[cargo_mgyp])
    except KeyError:
        return(None)

Next, let's get a list of each annotated cargo protein:

In [6]:
with open("../pfams/family_1_cargo_pfams.tsv", "r") as family1file:
    family1file.readline()
    family_1_pfams = [line.rstrip().split()[0] for line in family1file]

with open("../pfams/family_2_cargo_pfams.txt", "r") as family2file:
    family2file.readline()
    family_2_pfams = [line.rstrip().split()[0] for line in family2file]


all_pfams = family_1_pfams + family_2_pfams

annotated_cargos = set(pfam_df.query("Pfam in @all_pfams")["Cargo MGYP"].unique())

encapsulin_annotated_cargos = []

for cargo_mgyp in hits_df["Query"].unique():
    if get_encapsulin(cargo_mgyp) in annotated_encapsulins:
        encapsulin_annotated_cargos.append(cargo_mgyp)

encapsulin_annotated_cargos = set(encapsulin_annotated_cargos)

annotated_cargos = annotated_cargos.intersection(encapsulin_annotated_cargos)
len(annotated_cargos)

133

And now, let's get each one's best hit in the BLAST nr database:

In [7]:
hits_df = hits_df.query("Query in @annotated_cargos").sort_values(by="Identity", ascending=False).drop_duplicates(subset="Query").sort_values(by="Identity", ascending=True)
hits_df.iloc[:20]

Query                                             Target  \
174825  MGYP003302485827  PWL40683.1 hypothetical protein DBY43_06740 [C...   
370651  MGYP003626338685  CAB4162602.1 Glycosyl transferase, family 25 [...   
217548  MGYP001178528009  OUW91461.1 hypothetical protein CBD94_01720 [G...   
359273  MGYP003108877804  REK53366.1 hypothetical protein DWQ49_11805 [B...   
31578   MGYP003110663811  OUW91461.1 hypothetical protein CBD94_01720 [G...   
396388  MGYP001626144187  MCC6055797.1 hypothetical protein [Desulfuroco...   
175413  MGYP001626144193  MCC6041668.1 ferritin-like domain-containing p...   
73529   MGYP001996287958  GIR05032.1 MAG: hypothetical protein CM15mP16_...   
31436   MGYP001346362583  MBD1141706.1 4-hydroxybenzoate octaprenyltrans...   
37649   MGYP001438425319  MBD1141706.1 4-hydroxybenzoate octaprenyltrans...   
203556  MGYP001187095434  GIR05032.1 MAG: hypothetical protein CM15mP16_...   
320076  MGYP001772614563  MCS7128848.1 hypothetical protein [Sulfolobale...   
240218  MGYP003667212098  NIV76146.1 ferritin [Gammaproteobacteria bacte...   
377259  MGYP002732016615  MCH4154831.1 polyprenyl synthetase family prot...   
13336   MGYP003627660642  MCB1015492.1 ferritin [Acidimicrobiales bacter...   
84655   MGYP003143336435  WP_101934931.1 Dyp-type peroxidase [Mesorhizob...   
206826  MGYP000270495616  HAS11879.1 ferritin [Acidimicrobiaceae bacterium]   
326431  MGYP001442578972  MBC36368.1 4-hydroxybenzoate octaprenyltransfe...   
36013   MGYP001770665330  NAY81317.1 di-trans,poly-cis-decaprenylcistran...   
396077  MGYP003636907300        NBQ16825.1 hypothetical protein [bacterium]   

        Identity  Alignment Length  Mismatches  Gap Openings  Query Start  \
174825     0.327               159         101             3            9   
370651     0.504               228         102             4            7   
217548     0.553               177          78             1            1   
359273     0.559               177          77             1            1   
31578      0.564               177          76             1            1   
396388     0.617               290         111             0            1   
175413     0.619               155          59             0            1   
73529      0.634               230          84             0           54   
31436      0.636               154          56             0            1   
37649      0.636               154          56             0            1   
203556     0.636               154          56             0            1   
320076     0.666               156          51             1            1   
240218     0.669               127          41             1            1   
377259     0.673               322         105             0            1   
13336      0.684                95          30             0            8   
84655      0.704               301          89             0           23   
206826     0.744                94          24             0            4   
326431     0.746               276          70             0            6   
36013      0.758               265          64             0            1   
396077     0.759               162          39             0            3   

        Query End  Target Start  Target End        E-Value  Bitscore  
174825        161            12         170   6.246000e-12        76  
370651        228             2         224   3.018000e-68       242  
217548        176             1         177   8.637000e-53       195  
359273        176             1         177   1.047000e-54       200  
31578         176             1         177   1.303000e-53       197  
396388        290             1         290  1.246000e-104       351  
175413        155             1         155   2.187000e-56       204  
73529         283             1         230   3.108000e-80       280  
31436         154           130         283   7.383000e-56       202  
37649         154           130

These first few hits at very low identity look very interesting, let's take a look at their Pfam annotations:

In [8]:
low_identity_hits = hits_df.iloc[:20]["Query"].unique()
low_identity_pfams = pfam_df.query("`Cargo MGYP` in @low_identity_hits").sort_values(by="Cargo MGYP")
low_identity_pfams

Cargo MGYP     Pfam  Start  End  \
606615  MGYP000270495616  PF00619     15   58   
699138  MGYP000270495616   CL0044     21   40   
35754   MGYP001178528009  PF00210     36  164   
386260  MGYP001187095434  PF01040      3  148   
538789  MGYP001346362583  PF01040      3  148   
661325  MGYP001438425319  PF01040      3  148   
503401  MGYP001442578972  PF01040     22  282   
512447  MGYP001626144187  PF02915      4  125   
361212  MGYP001626144187  PF01988    118  281   
386483  MGYP001626144193  PF02915      7  144   
526854  MGYP001770665330  PF01255     36  266   
602219  MGYP001772614563  PF02915      7  144   
287734  MGYP001996287958  PF01040     17  280   
93082   MGYP002732016615  PF00348     28  279   
810834  MGYP003108877804  PF00210     37  134   
193320  MGYP003108877804  PF00210    138  163   
219057  MGYP003110663811  PF00210     37  122   
536769  MGYP003143336435  PF04261     12  324   
90444   MGYP003302485827  PF00210     74   94   
440437  MGYP003626338685  PF01977     31   52   
282242  MGYP003626338685  PF01755     11  189   
301967  MGYP003627660642   CL0044     20   49   
624618  MGYP003627660642   CL0041     13   60   
343087  MGYP003627660642   CL0041     65   86   
105933  MGYP003636907300  PF00210     10  156   
124990  MGYP003667212098  PF00619     15   93   
14920   MGYP003667212098   CL0044     19   47   

                                              Description  
606615                         Caspase recruitment domain  
699138                                           Ferritin  
35754                                Ferritin-like domain  
386260                      UbiA prenyltransferase family  
538789                      UbiA prenyltransferase family  
661325                      UbiA prenyltransferase family  
503401                      UbiA prenyltransferase family  
512447                                       Rubrerythrin  
361212                                         VIT family  
386483                                       Rubrerythrin  
526854         Putative undecaprenyl diphosphate synthase  
602219                                       Rubrerythrin  
287734                      UbiA prenyltransferase family  
93082                               Polyprenyl synthetase  
810834                               Ferritin-like domain  
193320                               Ferritin-like domain  
219057                               Ferritin-like domain  
536769                         Dyp-type peroxidase family  
90444                                Ferritin-like domain  
440437       3-octaprenyl-4-hydroxybenzoate carboxy-lyase  
282242  Glycosyltransferase family 25 (LPS biosynthesi...  
301967                                           Ferritin  
624618                                              Death  
343087                                              Death  
105933                               Ferritin-like domain  
124990                         Caspase recruitment domain  
14920                                            Ferritin

In [9]:
get_encapsulin("MGYP003302485827")

'MGYP003302485682'

## Interesting Hits

`MGYP003302485827` is the interesting example here - it's annotated as a Ferritin which is a known encapsulin however its best hit in the BLAST nr database is `PWL40683.1 hypothetical protein DBY43_06740 [Clostridiaceae bacterium]` with only 32% identity.

If we search its ESMFold predicted structure using Foldseek we get significant hits against crystal structures of ferritins from *E. coli* K12 (PDB codes **4ZTT** and **4XGS**).

Its encapsulin protein is `MGYP003302485682`, let's find out more about it:

In [10]:
uniref_hits_df = pd.read_csv("../encapsulin_UniRef90_hits.tsv", sep="\t", names=["Query", "Target", "Identity", "Alignment Length", "Mismatches", "Gap Openings",
                                                              "Query Start", "Query End", "Target Start", "Target End", "E-Value", "Bitscore"])

uniref_hits_df = uniref_hits_df[uniref_hits_df["Query"] == "MGYP003302485682"]
uniref_hits_df

Query                                             Target  \
837  MGYP003302485682  UniRef90_A0A330LPZ7 Lipoprotein n=1 Tax=Morite...   

     Identity  Alignment Length  Mismatches  Gap Openings  Query Start  \
837     0.462               359         191             1            5   

     Query End  Target Start  Target End       E-Value  Bitscore  
837        363             4         360  1.249000e-94       324

We can see that its best hit in UniRef90 is `UniRef90_A0A330LPZ7 Lipoprotein n=1 Tax=Moritella yayanosii TaxID=69539 RepID=A0A330LPZ7_9GAMM` with identity of only 46.2%